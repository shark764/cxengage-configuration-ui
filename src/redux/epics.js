import { combineEpics } from 'redux-observable';
import * as brandingEpics from './modules/entities/branding/epics';
import * as entitiesEpics from './modules/entities/epics';
import * as columnFiltersEpics from './modules/columnFilterMenus/epics';
import * as interactionMonitoringEpics from './modules/reporting/interactionMonitoring/epics';
import * as supervisorToolbarEpics from './modules/supervisorToolbar/epics';
import * as userIdMapEpics from './modules/userIdMap/epics';
import * as emailTemplatesEpics from './modules/entities/emailTemplates/epics';
import * as usersEpics from './modules/entities/users/epics';
import * as dataAccessReportsEpics from './modules/entities/dataAccessReports/epics';
import * as reasonListsEpics from './modules/entities/reasonLists/epics';
import * as flowsEpics from './modules/entities/flows/epics';
import * as dispatchMappingsEpics from './modules/entities/dispatchMappings/epics';
import * as slasEpics from './modules/entities/slas/epics';
import * as rolesEpics from './modules/entities/roles/epics';

export const rootEpic = combineEpics(
  ...Object.values(brandingEpics),
  ...Object.values(interactionMonitoringEpics),
  ...Object.values(entitiesEpics),
  ...Object.values(columnFiltersEpics),
  ...Object.values(supervisorToolbarEpics),
  ...Object.values(userIdMapEpics),
  ...Object.values(emailTemplatesEpics),
  ...Object.values(usersEpics),
  ...Object.values(dataAccessReportsEpics),
  ...Object.values(reasonListsEpics),
  ...Object.values(flowsEpics),
  ...Object.values(dispatchMappingsEpics),
  ...Object.values(slasEpics),
  ...Object.values(rolesEpics)
);
