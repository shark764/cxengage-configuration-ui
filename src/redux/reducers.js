/*
 * Copyright Â© 2015-2019 Serenova, LLC. All rights reserved.
 */

import { combineReducers } from 'redux-immutable';
import { reducer as form } from 'redux-form/immutable';

import Entities from './modules/entities';
import UserData from './modules/userData';
import SupervisorToolbar from './modules/supervisorToolbar';
import InteractionMonitoring from './modules/reporting/interactionMonitoring';
import AgentStateMonitoring from './modules/reporting/agentStateMonitoring';
import ColumnFilterMenus from './modules/columnFilterMenus';
import UserIdMap from './modules/userIdMap';

const reducer = combineReducers({
  Entities,
  UserData,
  form,
  SupervisorToolbar,
  InteractionMonitoring,
  AgentStateMonitoring,
  ColumnFilterMenus,
  UserIdMap
});

export default reducer;
