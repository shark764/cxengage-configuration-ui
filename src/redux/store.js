/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension/logOnlyInProduction';
import { createEpicMiddleware } from 'redux-observable';
import { rootEpic } from './epics';
import reducer from './reducers';

const composeEnhancers = composeWithDevTools({
  name: window.location.hash === '#/supervisorToolbar' ? 'Cx Supervisor Toolbar' : 'Cx Configuration'
});
export default createStore(reducer, composeEnhancers(applyMiddleware(createEpicMiddleware(rootEpic))));
