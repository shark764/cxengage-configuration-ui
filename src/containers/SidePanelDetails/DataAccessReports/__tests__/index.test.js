/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { Map } from 'immutable';
import { createStore } from 'redux';
import { shallow } from 'enzyme';
import DataAccessReportsDetailsPanel, { mapStateToProps } from '../';
import {
  getSelectedEntity,
  getCurrentEntity,
  userHasUpdatePermission,
  isInherited,
  isSaving
} from '../../../../redux/modules/entities/selectors';
import {
  getEntityListMembers,
  getListSize
} from '../../../../redux/modules/entities/listItemSelectors';
import {
  setSelectedSubEntityId,
  removeListItem
} from '../../../../redux/modules/entities';

jest.mock('../../../../redux/modules/entities/selectors');
getSelectedEntity.mockImplementation(
  () =>
    new Map({
      id: 'mockId',
      name: 'mockName',
      description: 'mockDescription',
      active: true,
      reportType: 'realtime',
      realtimeReportType: 'standard',
      realtimeReportName: 'IVR Interactions',
      members: [
        {
          id: 'mockSubEntityId'
        }
      ]
    })
);
getCurrentEntity.mockImplementation(() => 'dataAccessReports');
userHasUpdatePermission.mockImplementation(() => true);
isInherited.mockImplementation(() => false);
isSaving.mockImplementation(() => false);

jest.mock('../../../../redux/modules/entities/listItemSelectors');
getEntityListMembers.mockImplementation(() => ['mockListMembers']);
getListSize.mockImplementation(() => 0);

jest.mock('../../../../redux/modules/entities');
removeListItem.mockImplementation(() => 'mockId');
setSelectedSubEntityId.mockImplementation(() => 'mockId');

describe('DataAccessReportsDetailsPanel Renders', () => {
  it('renders', () => {
    const store = createStore(state => state);
    expect(
      shallow(
        <DataAccessReportsDetailsPanel store={store}>
          Child
        </DataAccessReportsDetailsPanel>
      )
    ).toMatchSnapshot();
  });
});

describe('Maps state to props only using selectors', () => {
  it('validates object created from mapStateToProps', () => {
    expect(mapStateToProps()).toMatchSnapshot();
  });
});
