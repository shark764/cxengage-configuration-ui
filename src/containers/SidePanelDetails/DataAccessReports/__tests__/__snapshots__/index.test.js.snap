// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataAccessReportsDetailsPanel Renders renders 1`] = `
ShallowWrapper {
  "length": 1,
}
`;

exports[`Maps state to props only using selectors validates object created from mapStateToProps 1`] = `
Object {
  "defaultFilters": Array [
    Object {
      "id": "invitationStatus",
      "value": "enabled",
    },
    Object {
      "id": "status",
      "value": "accepted",
    },
  ],
  "itemApiPending": undefined,
  "tableFields": Array [
    Object {
      "label": "Name",
      "name": "name",
    },
    Object {
      "label": "Email",
      "name": "email",
    },
    Object {
      "filterOptions": Array [
        "accepted",
        "pending",
        "invited",
        "disabled",
      ],
      "label": "Tenant Status",
      "name": "status",
      "type": "select",
    },
    Object {
      "filterOptions": Array [
        "pending",
        "invited",
        "expired",
        "enabled",
        "disabled",
        "sso-only",
      ],
      "label": "Platform Status",
      "name": "invitationStatus",
      "type": "select",
    },
  ],
  "tableItems": Array [
    "mockListMembers",
  ],
  "userHasUpdatePermission": true,
  "usersFetching": undefined,
}
`;
